### PR TITLE
[SELS-TASK][FE] Answering words

### PR DESCRIPTION
**Pull request by louisepujante**
### Asana
-[link](https://app.asana.com/0/1201426075515354/1201464141731489/f)

### Commands
-cd backend > composer install > php artisan migrate:fresh --seed
-cd frontend > npm install > npm start

### Pre-conditions
-copy backend\ .env.example and rename to backend.env then provide local db credentials

viewing categories list, answering categories, viewing category result:
-must be logged in and provide valid user credentials

### Expected Output
viewing categories list
-able to view all categories, able to start answering a category

answering categories
-able to choose an answer among the choices and click on next word

viewing category result
-able to view score, and if chosen answer is correct or not

### Screenshots
viewing categories list
image

answering categories
image
image
image

viewing category result
image

### CODE REVIEW

File: backend/app/Http/Controllers/Word/WordChoiceController.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    public function destroy(Word $word, Choice $choice)
+    {
+        $this->checkIfAdmin();
+        $this->checkChoice($word, $choice);
+        $choice->delete();
+        return $this->returnOne($choice);
+    }
+
+    protected function checkChoice(Word $word, Choice $choice) {
+        if($choice->word_id != $word->id) {
```
Comment (inahsanjuan) [inline backend/app/Http/Controllers/Word/WordChoiceController.php]:
```suggestion
        if ($choice->word_id != $word->id) {
```

File: backend/app/Http/Controllers/Word/WordChoiceController.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    public function update(Request $request, Word $word, Choice $choice)
+    {
+        $this->checkIfAdmin();
+        $this->checkChoice($word, $choice);
+        $request->validate([
+            'name' => ['required_without_all:is_correct', Rule::unique('choices', 'name')->ignore($choice->id)],
+            'is_correct' => ['required_without_all:name', 'boolean']
+        ]);
+
+        if($request['is_correct']) {
```
Comment (inahsanjuan) [inline backend/app/Http/Controllers/Word/WordChoiceController.php:64]:
```suggestion
        if ($request['is_correct']) {
```

File: backend/app/Http/Controllers/Word/WordChoiceController.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            $this->changePreviousCorrectAnswer($word);
+        } else {
+            //check if specified choice is the correct answer
+            $isCorrectAnswer = $choice->is_correct;
+            if($isCorrectAnswer) {
```
Comment (inahsanjuan) [inline backend/app/Http/Controllers/Word/WordChoiceController.php]:
```suggestion
            if ($isCorrectAnswer) {
```

File: frontend/src/admin/words/WordItem.tsx [hunk 4]
```diff
@@ -44,15 +43,21 @@ export default function WordItem(props: {
...
+                  return (
+                    <span
+                      className='md:text-2xl text-purple-400 font-semibold'
+                      key={choice.id}
+                    >
+                      {choice.name}
+                    </span>
+                  );
+                }
+              })}
```
Comment (inahsanjuan) [inline frontend/src/admin/words/WordItem.tsx]: Revert back to original code. Usa the `answer` variable

File: frontend/src/icons/ArrowLeftIcon.tsx [hunk 5]
```diff
@@ -1,11 +1,11 @@
 import React from 'react'
 
-export default function BackIcon(props: { className?: string }) {
+export default function ArrowLeftIcon(props: { className?: string }) {
   return (
     <svg
       xmlns='http://www.w3.org/2000/svg'
       version='1.1'
-      xmlnsXlink='http://www.w3.org/1999/xlink'
+      // xmlnsXlink='http://www.w3.org/1999/xlink'
```
Comment (inahsanjuan) [inline frontend/src/icons/ArrowLeftIcon.tsx]: Delete commented line

File: frontend/src/icons/ErrorIcon.tsx [hunk 6]
```diff
@@ -1,28 +1,28 @@
...
-                d='M14.5,25h3v-3h-3V25z M14.5,6v13h3V6H14.5z'
-                id='Exclamatory_x5F_Sign'
-                fill='#E6E6E6'
-              />
-            </g>
+export default function ErrorIcon(props: { className?: string }) {
+  return (
+    <svg
+      height='128'
+      // style='overflow:visible;enable-background:new 0 0 32 32'
```
Comment (inahsanjuan) [inline frontend/src/icons/ErrorIcon.tsx]: Delete commented line

File: frontend/src/user/categories/AnsweringCategory.tsx [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+                }`}
+                disabled={
+                  currentAnswer?.id === 0 || !currentAnswer ? true : false
+                }
+                onClick={() => handleNextWord()}
+              >
+                <span className='text-center'>
+                  {index + 1 < state.words.words.length
+                    ? 'Next word'
+                    : 'Finish category'}
```
Comment (inahsanjuan) [inline frontend/src/user/categories/AnsweringCategory.tsx]:
```suggestion
                  {
                    index + 1 < state.words.words.length
                      ? 'Next word'
                      : 'Finish category'
                  }
```

File: frontend/src/user/categories/CategoryResults.tsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+            <span className='text-2xl self-end'>
+              Score:{' '}
+              <span className='text-3xl text-purple-500 font-bold'>
+                {score.length}
+              </span>{' '}
+              out of {state.words.words.length}
+            </span>
+            {state.words.words.map((word) => {
+              return <WordItemResult word={word} />;
+            })}
```
Comment (inahsanjuan) [inline frontend/src/user/categories/CategoryResults.tsx]: Can make it one line
```suggestion
            {state.words.words.map((word) => <WordItemResult word={word} />)}
```

**Review by inahsanjuan — CHANGES_REQUESTED** [review]

File: frontend/src/user/categories/WordItemResult.tsx [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+                      ${
+                        answer?.id === choice.id &&
+                        !choice.is_correct &&
+                        'bg-failed'
+                      }
+                    `}
+                  >
+                    {choice.name}
+                  </span>
+                  {answer?.id === choice.id && (
```
Comment (inahsanjuan) [inline frontend/src/user/categories/WordItemResult.tsx]: remove the `0` under the word
image

**Review by inahsanjuan — CHANGES_REQUESTED** [review]

File: frontend/src/admin/words/WordItem.tsx (hunk 4 above)
Comment (louisepujante) [reply frontend/src/admin/words/WordItem.tsx]: na sayop siguro ni ug merge katung pag conflict miss sorry

**Pull request merged**